### PR TITLE
model_checkpoint to save all models (where validation occurs)

### PR DESCRIPTION
**Pull request by jamesjjcondon**
# Before submitting

- [x] Was this discussed/approved via a Github issue? #596 
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Improves some use-cases where saving all models is desired. See #596 .
Removes deleting of models on existing save filepath and drops delete warning if save_top_k == -1.
Enables saving of all epochs where validation occurs.
Does not address saving all epochs regardless of check_val_every_n_epochs > 1

Not tested with NVIDIA/apex

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Yes, but 2nd ever contribution (be gentle : )

### CODE REVIEW

**Comment by Borda** [comment]
this failing shall be fixed in #1163

**Comment by Borda** [comment]
@jamesjjcondon  could you pls update from master? we have fixed the failier in #1163

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def __init__(self, filepath, monitor: str = 'val_loss', verbose: bool = False,
                  save_top_k: int = 1, save_weights_only: bool = False,
                  mode: str = 'auto', period: int = 1, prefix: str = ''):
         super().__init__()
-        if save_top_k and os.path.isdir(filepath) and len(os.listdir(filepath)) > 0:
+        if save_top_k > 0 and os.path.isdir(filepath) and len(os.listdir(filepath)) > 0:
```
Comment (jeremyjordan) [inline pytorch_lightning/callbacks/model_checkpoint.py]: what's the motivation for dropping the delete warning if `save_top_k == -1`?

**Comment by MattPainter01** [comment]
Seems I can't push to this PR (permission denied) to merge in master. 

@jamesjjcondon If you could merge it in that would be great.

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 1 above)
Comment (MattPainter01) [reply pytorch_lightning/callbacks/model_checkpoint.py]: Looks to me that we only call ``_del_model`` when save_top_k > 0 (lines 225, 226), so I think this is fine currently. We could clear it once if we're saving every epoch though, not sure if that's preferable.

**Comment by Borda** [comment]
@jamesjjcondon how is it going? could we get this done for next release next week?

**Comment by williamFalcon** [comment]
Finishing in #1359

**Pull request closed without merging**